shouldupdate.js

```diff
@@ -236,6 +184,22 @@ function factory (methods) {
   }
 }
 
+function compare (current, next, comparator, equalCheck) {
```

Comment:
Just a nitpick. I think `comparator` is the wrong name. 

Maybe rename to `typecheck` or similar.